File: 2a_InterfacePattern/src/Car.java
public class Car implements Vehicle {

    @Override
    public int set_num_of_wheels() {
        return 4; //Usual cases for most cars
    }

    @Override
    public int set_num_of_passengers() {
        return 4; //If it's a normal 4 seated private cars
    }

    @Override
    public boolean has_gas() {
        return true; //Car runs on Octane gas
    }


}

File: 2a_AbstractFactoryPattern/src/Car.kt
class Car : Vehicle() {
    override fun set_num_of_wheels(): Int {
        return 4//Usual cases for most cars
    }

    override fun set_num_of_passengers(): Int {
        return 4//If it's a normal 4 seated private cars
    }

    override fun has_gas(): Boolean {
        return true //Car runs on Octane gas
    }
}